tests/integration/test_utils/data_source_config/base.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import random
+import string
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from functools import cached_property
+from typing import TYPE_CHECKING, Generic, Optional, TypeVar
+
+import great_expectations as gx
+from great_expectations.data_context.data_context.abstract_data_context import AbstractDataContext
+from great_expectations.datasource.fluent.interfaces import Batch
+
+if TYPE_CHECKING:
+    import pandas as pd
+    import pytest
+
+
+@dataclass(frozen=True)
+class DataSourceConfig(ABC):
```

Comment:
Nit - could we update this name and the one below to signify this is a config for a test? Just to differentiate between an actual DS config

Reply:
Yeah, I like that!